### PR TITLE
feat(deal-ticket): calculate fees and discount base on volumeDiscountStats and referralSetStats

### LINKED ISSUE #4714: See volume discounts in the deal ticket
## Story

As a user connected with a key that is part of a referral code / team
I want to see the discount I will receive on my trading fees reflected in the deal ticket
So that I understand the fee I will be paying, but can see the discount applied

## Acceptance Criteria

- [ ] I can see what my volume discount factor is
- [ ] I can see the estimated fee I will pay for a position with my discount applied (calculate in client)
- [ ] I can see what the trading fee would have been otherwise 
- [ ] ~The total discount is volume discount + referral discount (NOT multiplied!)~
- [ ] The fully_discounted_fee as defined below is displayed on the ticket 
- [ ] The discount_factor as defined below is displayed on the ticket

Given you get `estimatedFee` from the API

```
referralDiscount = floor(referral_discount_factor * estimatedFee)
discounted_fee = fee - referralDiscount

then 

volumeDiscount = floor(volume_discount_factor * discounted_fee)
fully_discounted_fee = discounted_fee - volume_discount

and 

discount_factor = fully_discounted_fee/fee
```

## Tasks

- [ ] UX (if needed)
- [ ] Design (if needed)
- [ ] Team and stakeholder review
- [ ] Specs reviewed and created or adjusted
- [ ] Implementation
- [ ] Testing (unit and/or e2e)
- [ ] Code review
- [ ] QA review

## Sketch

Shoudl already have a pattern for this established based on the referral ticket #4408 

Image

https://miro.com/app/board/uXjVOs_PBKk=/?moveToWidget=3458764559935654833&cot=14

## Additional details / background info

The factors applied can be found in this ticket:

https://github.com/vegaprotocol/vega/issues/9258

They will be provided on a new API to be built.

### PR DESCRIPTION
**Pull request by bglownia**
# Related issues 🔗

Closes #4714 

# Description ℹ️

Shows fees with related volume and referral discounts

# Demo 📺

Screenshot 2023-10-30 at 16 47 07

### CODE REVIEW

File: libs/markets/src/lib/market-utils.ts [hunk 1]
```diff
@@ -51,15 +51,17 @@ export const getQuoteName = (market: Partial<Market>) => {
 
 export const sumFeesFactors = (fees: Market['fees']['factors']) => {
   return fees
-    ? new BigNumber(fees.makerFee)
-        .plus(fees.liquidityFee)
-        .plus(fees.infrastructureFee)
+    ? Number(fees.makerFee) +
```
Comment (MadalinaRaicu) [inline libs/markets/src/lib/market-utils.ts:54]: why this change was needed?

**Review by MadalinaRaicu — APPROVED** [review]
LGTM. some extra details on the PR would be helpful for review, maybe a description or screenshots if applicable

File: libs/markets/src/lib/market-utils.ts (hunk 1 above)
Comment (mattrussell36) [reply libs/markets/src/lib/market-utils.ts:54]: This would be better left as BigNumber to avoid a floating point error..

File: libs/deal-ticket/src/components/fees-breakdown/fees-breakdown.tsx [hunk 2]
```diff
@@ -43,85 +43,95 @@ const FeesBreakdownItem = ({
...
-  if (totalFees === '0') return null;
+  if (!fees || !totalFeeAmount || totalFeeAmount === '0') return null;
+
+  const { discountedFee: discountedInfrastructureFee } = getDiscountedFee(
+    fees.infrastructureFee,
+    referralDiscountFactor,
+    volumeDiscountFactor
+  );
+
+  const { discountedFee: discountedLiquidityFee } = getDiscountedFee(
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/fees-breakdown/fees-breakdown.tsx:70]: Might as well rename these as useDiscountedFee so we get a better experience in react dev tools

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/markets/src/lib/market-utils.ts (hunk 1 above)
Comment (mattrussell36) [reply libs/markets/src/lib/market-utils.ts]: I've updated it to use BigNum

File: libs/deal-ticket/src/components/fees-breakdown/fees-breakdown.tsx (hunk 2 above)
Comment (mattrussell36) [reply libs/deal-ticket/src/components/fees-breakdown/fees-breakdown.tsx:70]: Nevermind on this one. Bigger changes would be required due to conditional rendering of hooks

**Review by mattrussell36 — APPROVED** [review]
